Feat/deploy workflow

wottpal opened the pull request.

wottpal commented:
This pull request has been deployed to Vercel.

<table>
<tr>
<td><strong>Latest commit:</strong></td>
<td><code>8eee71e</code></td>
</tr>
<tr>
<td><strong>✅ Preview:</strong></td>
<td><a href='https://yieldgate-lk3k6j11g-dennis-zoma.vercel.app'>https://yieldgate-lk3k6j11g-dennis-zoma.vercel.app</a></td>
</tr>
<tr>
<td><strong>🔍 Inspect:</strong></td>
<td><a href='https://vercel.com/dennis-zoma/yieldgate/EF9vFVacv2HV4NkAa1qjmry9eMwo'>https://vercel.com/dennis-zoma/yieldgate/EF9vFVacv2HV4NkAa1qjmry9eMwo</a></td>
</tr>
</table>

[View Workflow Logs](https://github.com/yieldgate/yieldgate/actions/runs/2454801937)

sebastianst commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,26 @@
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: [develop]
```
IIRC we wanted to remove these lines so we only trigger it manually. We can leave the `push` hook and see what happens :)

sebastianst replied:
Actually, we probably also want to delete the `push` hook because vercel should already watch the `develop` branch.

wottpal replied:
> because vercel should already watch the develop branch. 


I don't think it does, because we need Pro to work with orga-repos (which is one of the reasons for this CI-workflow in the first place). Or do I miss anything?

wottpal commented:
Little update from my side:
* Added @yield-e to organization & project
* Updated project secrets accordingly (Vercel-, and Github-tokens)
* Created & configured new Vercel-Account with @yield-e and setup Deployment/Environment (got a manual deploy running through even, see https://ygyg.vercel.app/)

**But**: The deploy workflow doesn't run through now :( Immediately errors with an error-message I couldn't really decipher/debug. I created an issue here: https://github.com/BetaHuhn/deploy-to-vercel-action/issues/187

In the meantime, I'll further improve the script to also cover production deploys.

(Also FYI a little tip: after re-adding the `pull_request: branches: [develop]` to the script for debugging i found out that i can already trigger the script from my local terminal without having the trigger-button visible yet: `gh workflow run deploy.yml --ref feat/deploy-workflow`)

wottpal commented:
* Added different deploy steps for different environments:
   → `pr-{PR}.ygyg.vercel.app` for manual workflow dispatches
   → `staging.ygyg.vercel.app` for pushes on `develop`
   → `prod.ygyg.vercel.app` for pushes on `main` (`yieldgate.xyz` will be another alias when domain is moved)

wottpal commented:
@BetaHuhn thanks for the quick reply! Yes the error is still coming up unfortunately.

sebastianst commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,50 @@
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, '[skip ci]')"
```
Is this `if` condition still needed, now that we manually trigger the deployment?

wottpal replied:
it could be used if we attach `[skip ci]` to the merge commit i guess – if we don't want to update staging/production. don't know how often this will be the case though.

sebastianst replied:
fair enough, doesn't hurt either to have that option

wottpal commented:
Changed the domain alias scheme as follows (vercel does not seem to like subdomains):
* Preview: `ygyg-{SHA}.vercel.app`
* Staging/Develop: `ygyg-dev.vercel.app`
* Prod: `ygyg-main.vercel.app`

airtonix commented:
simply changing the version to `1.9.0` doesn't work for the rest of us. is there something else you guys are doing to make this work ?

wottpal commented:
> simply changing the version to `1.9.0` doesn't work for the rest of us. is there something else you guys are doing to make this work ?

Hey @airtonix, if I remember correctly that was what fixed the issue for us 🤷‍♂️

You can have a look at our deploy workflow here: https://github.com/yieldgate/yieldgate/blob/develop/.github/workflows/deploy.yml

Sorry for not being more helpful..

airtonix commented:
this is my workflow that is struggling: 

```
name: Release

on:
  push:
    branches:
      - master

concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true

jobs:
  Release:
    name: Release Packages and Changelogs

    runs-on: ubuntu-22.04

    permissions:
      contents: write
      pull-requests: write
      issues: write
      actions: write
      deployments: write
      packages: write

    steps:
      - uses: actions/checkout@v3
        with:
          # We need to fetch all branches and commits so that Nx affected has a base to compare against.
          fetch-depth: 0
          clean: false

      - name: Derive appropriate SHAs for base and head for `nx affected` commands
        uses: nrwl/nx-set-shas@87839d6fba2ae7c5c7fd235475c0a59c58099fb8
        with:
          main-branch-name: master

      - uses: ./pkg/actions/setup-tooling
      - uses: ./pkg/actions/setup-yarn

      - name: Build a list of affected packages
        uses: dkhunt27/action-nx-affected-list@66859c9e82c136ecbb996881ba213bbc03f5a5c3
        id: affected
        with:
          base: ${{ env.NX_BASE }}
          head: ${{ env.NX_HEAD }}

      - run: yarn nx affected --target=semantic-release --maxParallel=1
        env:
          GITHUB_TOKEN: ${{ secrets.BOT_TOKEN }}
          GIT_AUTHOR_EMAIL: ${{secrets.BOT_GIT_EMAIL}}
          GIT_AUTHOR_NAME: ${{secrets.BOT_GIT_NAME}}
          GIT_COMMITTER_EMAIL: ${{secrets.BOT_GIT_EMAIL}}
          GIT_COMMITTER_NAME: ${{secrets.BOT_GIT_NAME}}

      - name: Deploy App:Storybook to Vercel
        if: contains(steps.affected.outputs.affected, 'storybook')
        uses: BetaHuhn/deploy-to-vercel-action@b3d098918bfd16ccb8519fe5b2a43446e86987f4
        with:
          GITHUB_TOKEN: ${{ secrets.BOT_TOKEN }}
          VERCEL_TOKEN: ${{ secrets.VERCEL_TOKEN }}
          VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
          VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID_STORYBOOK }}
          GITHUB_DEPLOYMENT_ENV: storybook_production

      - name: Deploy App:Dashboard to Vercel
        if: contains(steps.affected.outputs.affected, 'dashboard')
        uses: BetaHuhn/deploy-to-vercel-action@b3d098918bfd16ccb8519fe5b2a43446e86987f4
        with:
          GITHUB_TOKEN: ${{ secrets.BOT_TOKEN }}
          VERCEL_TOKEN: ${{ secrets.VERCEL_TOKEN }}
          VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
          VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID_DASHBOARD }}
          GITHUB_DEPLOYMENT_ENV: dashboard_production

```

- `BetaHuhn/deploy-to-vercel-action@b3d098918bfd16ccb8519fe5b2a43446e86987f4` is the ref to `BetaHuhn/deploy-to-vercel-action@v1.9.0`


but on my PR workflow: no problems...

```
name: CI

on:
  pull_request:
    branches:
      - master

concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true

jobs:
  Review:
    name: Verify
    runs-on: ubuntu-22.04

    permissions:
      contents: read
      actions: write
      pull-requests: read
      checks: read

    steps:
      - uses: actions/checkout@v3
        with:
          # By default, PRs will be checked-out based on the Merge Commit, but we want the actual branch HEAD.
          ref: ${{ github.event.pull_request.head.sha }}
          # We need to fetch all branches and commits so that Nx affected has a base to compare against.
          fetch-depth: 0
          clean: false

      - name: Derive appropriate SHAs for base and head for `nx affected` commands
        uses: nrwl/nx-set-shas@v2
        with:
          main-branch-name: master

      - uses: ./pkg/actions/setup-tooling
      - uses: ./pkg/actions/setup-yarn

      - name: Build a list of affected packages
        uses: dkhunt27/action-nx-affected-list@66859c9e82c136ecbb996881ba213bbc03f5a5c3
        id: affected
        with:
          base: ${{ env.NX_BASE }}
          head: ${{ env.NX_HEAD }}

      - run: yarn nx workspace-lint
      - run: yarn nx affected --target=lint --parallel=3
      - run: yarn nx affected --target=test --parallel=3 --ci --code-coverage
      - run: yarn nx affected --target=build --parallel=3

      - name: Deploy App:Storybook to Vercel
        if: contains(steps.affected.outputs.affected, 'storybook')
        uses: BetaHuhn/deploy-to-vercel-action@b3d098918bfd16ccb8519fe5b2a43446e86987f4
        with:
          GITHUB_TOKEN: ${{ secrets.BOT_TOKEN }}
          VERCEL_TOKEN: ${{ secrets.VERCEL_TOKEN }}
          VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
          VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID_STORYBOOK }}
          GITHUB_DEPLOYMENT: false

      - name: Deploy App:Dashboard to Vercel
        if: contains(steps.affected.outputs.affected, 'dashboard')
        uses: BetaHuhn/deploy-to-vercel-action@b3d098918bfd16ccb8519fe5b2a43446e86987f4
        with:
          GITHUB_TOKEN: ${{ secrets.BOT_TOKEN }}
          VERCEL_TOKEN: ${{ secrets.VERCEL_TOKEN }}
          VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
          VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID_DASHBOARD }}
          GITHUB_DEPLOYMENT: false

```

sebastianst commented:
Would be good to continue this conversation in an issue/PR of the affected repo. This has nothing to do with this PR here :)
